src/quo/components/wallet/approval_label/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns quo.components.wallet.approval-label.view
+  (:require [oops.core :refer [oget]]
+            [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icon]
+            [quo.components.markdown.text :as text]
+            [quo.components.wallet.approval-label.schema :as approval-label.schema]
+            [quo.components.wallet.approval-label.style :as style]
+            [quo.foundations.colors :as colors]
+            quo.theme
+            [react-native.core :as rn]
+            [react-native.hole-view :as hole-view]
+            [schema.core :as schema]
+            [utils.i18n :as i18n]))
+
+(def ^:private status-icons
+  {:approve   :i/alert
+   :approving :i/pending-state
+   :approved  :i/check})
+
+(def ^:private status-message
+  {:approve   :t/approve-amount-symbol
+   :approving :t/approving-amount-symbol
+   :approved  :t/approved-amount-symbol})
+
+(defn- view-internal
+  [{:keys [status customization-color token-value token-symbol
+           container-style button-props]
+    :or   {customization-color :blue}}]
+  (let [theme                                 (quo.theme/use-theme)
+        [container-width set-container-width] (rn/use-state 0)
+        on-layout                             (rn/use-callback #(set-container-width
+                                                                 (oget % :nativeEvent :layout :width)))]
+    [hole-view/hole-view
+     {:holes     [{:x                       0
+                   :y                       0
+                   :height                  24
+                   :width                   container-width
+                   :borderBottomStartRadius 16
+                   :borderBottomEndRadius   16}]
+      :style     {:margin-top -24}
+      :on-layout on-layout}
+     [rn/view
+      {:style               (merge (style/container customization-color theme)
+                                   container-style)
+       :accessibility-label :approval-label}
+      [rn/view {:style style/content}
+       [icon/icon
+        (status status-icons)
+        {:color (colors/theme-colors colors/neutral-50 colors/neutral-40 theme)
+         :size  16}]
+       [text/text
+        {:size  :paragraph-2
+         :style (style/message theme)}
+        (i18n/label (status status-message)
```

Comment:
Nitpick: it's quite common in Clojure that if the map is statically defined, thus can never, ever be nil, the args are reversed. In other words, the static map would be treated as a function and become the first argument. So reversed it would be `(status-message status)`.

Same logic can be applied to `(status-icons status)`.

Reply:
I have updated the code. Is there any benefit to doing it like this?

Reply:
Thanks for the explanation!